packages/htmlbars-syntax/lib/parser/tokenizer-event-handlers.js

```diff
@@ -200,6 +202,17 @@ function assembleConcatenatedValue(parts) {
   return b.concat(parts);
 }
 
+function validateStartTag(tag, tokenizer) {
+  var error;
+
+  // no support for <script> tags
+  if (tag.name === "script") {
+    error = "SCRIPT tags are not allowed in HTMLBars templates (line " + tokenizer.tagLine + ")";
```

Comment:
I agree with @stefanpenner, we should change both the validation functions to simply throw immediately.